client/reader/site-stream/featured.jsx

```diff
@@ -24,8 +26,37 @@ export default React.createClass( {
 
 	getStateFromStores( store ) {
 		store = store || this.props.store;
+
+		let postKeys = store.get();
+		let posts = postKeys.map( postKey => {
+			let post = FeedPostStore.get( postKey ),
+				originalPost,
+				isDiscoverPost;
+
+			if ( ! post || post._state === 'minimal' ) {
+				FeedPostStoreActions.fetchPost( postKey );
+			} else {
+				isDiscoverPost = post && DiscoverHelper.isDiscoverPost( post );
+
+				let isDiscoverSitePick = post && isDiscoverPost && DiscoverHelper.isDiscoverSitePick( post );
+
+				if ( isDiscoverPost && ! isDiscoverSitePick && get( post, 'discover_metadata.featured_post_wpcom_data.blog_id' ) ) {
+					originalPost = FeedPostStore.get( {
+						blogId: post.discover_metadata.featured_post_wpcom_data.blog_id,
+						postId: post.discover_metadata.featured_post_wpcom_data.post_id
+					} );
+				}
+			}
```

Comment:
I'm pretty confused what the purpose of this block is for, but I think that if we had a helper function for the conditional it would be clearer

```js
getStateFromStores( store ) {
	const needsOriginal = post => {
		if ( ! post ) {
			return false;
		}

		const isDiscoverPost = DiscoverHelper.isDiscoverPost( post );
		const isDiscoverSitePick = isDiscoverPost && DiscoverHelper.isDiscoverSitePick( post );
		return ! isDiscoverSitePick && get( post, 'discover_metadata.featured_post_wpcom_data.blog_id' );
	};

	const shouldFetch = post => ! post || post._state === 'minimal';

	const posts = store.get().map( postKey => {
		const post = FeedPostStore.get( postKey );

		if ( shouldFetch( post ) ) {
			FeedPostStoreActions.fetchPost( postKey );
			return { post };
		}

		if ( needsOriginal( post ) ) {
			const { blog_id: blogId, post_id: postId } = post.discover_metadata.featured_post_wpcom_data;

			return {
				post
				orignalPost: FeedPostStore.get( { blogId, postId } ),
				isDiscoverPost: true
			}
		}
	} );

	return { posts };
}
```